middleware/src/middleware.go

```diff
@@ -913,6 +908,43 @@ func (middleware *Middleware) SetLoginPassword(args rpcmessages.SetLoginPassword
 	}
 }
 
+// FinalizeSetupWizard finalizes the setup wizard by setting a redis key,
+// enabling bitcoin core and relates services and starting bitcoin code and
+// related services.
+func (middleware *Middleware) FinalizeSetupWizard() rpcmessages.ErrorResponse {
+	log.Println("Finalizing the setup wizard.")
+
+	err := middleware.redisClient.SetString(redis.BaseSetupDone, "1")
```

Comment:
I think it would make sense to set the Redis key after the fact, to store success. If this RPC fails, it might not be called again although the services would stay disabled.

Reply:
Agree